### ISSUE TITLE
how to do ?

### ISSUE THREAD

**Issue by wxlg1117** [andyneff/esxi-nut#1]
git clone https://github.com/andyneff/docker-esxi.git
docker build --build-arg ISO_IMAGE=vm6.iso -t andyneff/esxi .

git clone https://github.com/andyneff/esxi-nut.git
cd esxi-nut/
docker build -t andyneff/esxi-nut .
?

How can I get nut vib file?

**Comment by andyneff** [comment]
Ah, this project does need a README

I believe you can either

```
docker run -it --rm -v `pwd`:/data andyneff/esxi-nut
```

Or 

```
docker-compose run nut
```

And then you should see a `NutClient-ESXi-${NUT_CLIENT_VERSION}.tar.gz` file, probably owned by root :(

**Comment by wxlg1117** [comment]
after run "docker run -it --rm -v `pwd`:/data andyneff/esxi-nut ", get some error ...

root@nodeovh1:~# docker run -it --rm -v `pwd`:/data andyneff/esxi-nut
tar: can't open '/data/vib_ar/upsmon': No such file or directory
sha256sum: can't open '/data/vib_ar/upsmon': No such file or directory
zcat: /data/vib_ar/upsmon: No such file or directory
stat: can't stat '/data/vib_ar/upsmon': No such file or directory
/bin/sh: can't create /data/vib_ar/descriptor.xml: nonexistent directory
/bin/sh: can't create /data/package/upsmon-install.sh: nonexistent directory
/bin/sh: can't create /data/package/upsmon-update.sh: nonexistent directory
chmod: /data/package/upsmon-install.sh: No such file or directory
chmod: /data/package/upsmon-update.sh: No such file or directory
/bin/sh: cd: line 1: can't cd to /data/vib_ar
ar: /data/package/upsmon-2.7.4-2.0.1.x86_64.vib: No such file or directory
/bin/sh: cd: line 1: can't cd to /data/package
tar: *.txt: No such file or directory
tar: *.sh: No such file or directory
tar: upsmon-2.7.4-2.0.1.x86_64.vib: No such file or directory
tar: error exit delayed from previous errors

**Comment by andyneff** [comment]
If you are doing to use the `` `pwd` `` notation, you need to be IN the repo directory.

```
docker run -it --rm -v {the directory of the esxi-nut repo}:/data andyneff/esxi-nut
```

Please put the curly brace with the directory of the esxi-nut repo

**Comment by jeykaliraj** [comment]
Dear Andy, What is the difference between both docker-esxi and esxi-nut project. Is it possible to dockerize even the latest esxi-6.7u3 version and run as container. Any response on this is much appreciated. I am really passionate about this project.

**Comment by wxlg1117** [comment]
@jeyakumarkaliraj, docker-esxi is is a compilation environment ,esxi-nut contains many configurations for nut repo ....

**Comment by andyneff** [comment]
@jeyakumarkaliraj, wxlg1117 is right.

`docker-esxi` is a docker image of ESXi based on the ESXi iso. The purpose of this image is to give you a docker environment to build ESXi plugins in, instead of using another OS that is "close enough" and hope it works. Using this, we know it will work (unless there are differences the CPU/kernel that affect how something is compiled, which can usually be fixed with compiler flags)

Since I do not distribute these images, you have to build them yourself from the ISO, which I try to make easy. You should be able to just use the esxi-6.7u3 iso, but I haven't used this in a while.

exi-nut builds on top of that, to build the Network UPS Tools for that version of ESXi. So this is one such example of building a plugin for ESXi.

FYI: In the ESXi 6.7, I was unable to get mutt to function properly, so email notifications were not working. I could not determine the cause of this, everything seems to be configured properly, but just won't email.

**Comment by jeykaliraj** [comment]
Dear Andy - I performed the following steps and I could login to esxi shell prompt. 
I could find the ipaddress of esxi (172.17.0.2) from docker inspect container name. While I try to access the esxi console from https://172.17.0.2, connection refused

git clone https://github.com/andyneff/docker-esxi.git
docker build --build-arg ISO_IMAGE=vm6.iso -t andyneff/esxi .
docker run -it --rm -v $PWD:/data andyneff/esxi

Can you please help me how to run esxi as docker container

**Comment by andyneff** [comment]
TL;DR I'm afraid that docker can't do what you want it to do.

@jeyakumarkaliraj Unfortunately you cannot run esxi as a container.

> The purpose of this image is to give you a docker environment to build ESXi plugins in.

ESXi is a type 1 VM hypervisor. This means it run on the "bare metal" of a computer, not inside of an OS or VM or Container. (Maybe there is a way to trick a VM to run esxi, I don't know, but that would have nothing to do with docker or this image.) What you are suggesting is just incompatible with that the technology does.

- This image does not run any of the esxi services. 
- I suppose in theory if you are trying to develop a mod for the web UI, you could try and start the webserver, but I don't remember how to start services in ESXi, or even if that is possible. But it does not sound like this is what you are trying to do

---

When you compile code on a Linux machine, that code "links" with all the libraries of that operating system. So when you transfer that compiled binary from one computer to the next, it only works perfectly if the library it "linked" to on the compiled machine are compatible with the libraries on the destination machine.

Historically, this was done for ESXi by saying "I will compile on CentOS/RHEL 5, where the libraries are **so** old, that newer libraries are _usually_ backwards compatible.

I wanted to take this a step further, and make no assumptions about compatibility, so I made `docker-esxi` that will have the exactly correct libraries for you to link against, so when you transfer the plugin file, I knew it would work.

---

If you have further question, please open your own issue on https://github.com/andyneff/docker-esxi
